src/modules/party/dto/create-party-request.dto.ts

```diff
@@ -1,62 +1,76 @@
-import { ApiProperty } from '@nestjs/swagger';
-import { EXAMPLES, PROPERTIES } from '@ukef/constants';
-import { IsIn, IsISO8601, IsOptional, IsString, Length, Matches } from 'class-validator';
+import { ENUMS, EXAMPLES, PROPERTIES } from '@ukef/constants';
+import { ValidatedDateOnlyApiProperty } from '@ukef/decorators/validated-date-only-api-property.decorator';
+import { ValidatedStringApiProperty } from '@ukef/decorators/validated-string-api-property.decorator';
 
-export type CreatePartyRequest = CreatePartyRequestItem[];
+export type CreatePartyRequestDto = CreatePartyRequestItem[];
 
 export class CreatePartyRequestItem {
-  @ApiProperty({ description: 'The UKEF ID for the party. Should contain just digits.', minLength: 8, maxLength: 8, example: '00291013', pattern: '/^d{8}$/' })
-  @Length(8, 8, {
-    message: 'alternateIdentifier must be exactly 8 characters',
-  })
-  @Matches(/^\d+$/, {
-    message: 'alternateIdentifier must only contain digits',
+  @ValidatedStringApiProperty({
+    description: 'The UKEF ID for the party.',
+    length: 8,
+    example: EXAMPLES.PARTY_ID,
   })
   alternateIdentifier: string;
 
-  @ApiProperty({ description: 'The primary industry classification code for this customer.', minLength: 1, maxLength: 10, example: '0001' })
-  @Length(1, 10)
+  @ValidatedStringApiProperty({
+    description: 'The primary industry classification code for this customer.',
+    minLength: 1,
+    maxLength: 10,
+    example: EXAMPLES.INDUSTRY_CLASSIFICATION,
+  })
   industryClassification: string;
 
-  @ApiProperty({ description: 'The primary customer name.', minLength: 1, maxLength: 35, example: 'ACTUAL IMPORT EXPORT' })
-  @Length(1, 35)
+  @ValidatedStringApiProperty({
+    description: 'The primary customer name.',
+    minLength: 1,
+    maxLength: 35,
+    example: EXAMPLES.PARTY_NAME,
+  })
   name1: string;
 
-  @ApiProperty({ description: 'The secondary customer name.', required: false, minLength: 1, maxLength: 35, example: 'ACTUAL IMPORT EXPORT' })
-  @IsOptional()
-  @Length(0, 35)
+  @ValidatedStringApiProperty({
+    description: 'The secondary customer name.',
+    required: false,
+    maxLength: 35,
+    example: EXAMPLES.PARTY_NAME,
+  })
   name2: string;
```

Comment:
(I know it's not your fault but) the `required: false` fields should really be optional in our type system as well (e.g. `name2?: string`)

Reply:
Added `?` to all the `required: false` fields in this DTO